File: src/0392.is-subsequence.392/solution.ts
function isSubsequence (s: string, t: string): boolean {
    let indexS = 0;
    let indexT = 0;
    while (indexS < s.length && indexT < t.length) {
        if (s[indexS] === t[indexT]) {
            indexS++;
            indexT++;
        } else {
            indexT++;
        }
    }
    return indexS === s.length;
}

File: src/0392.is-subsequence.392/solution.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        char[] sc = s.toCharArray();
        char[] tc = t.toCharArray();
        int i = 0;
        int j = 0;

        while (i < sc.length && j < tc.length) {
            if (sc[i] != tc[j]) {
                j++;
            } else {
                i++;
                j++;
            }
        }
        return i == sc.length;
    }
}